types/src/contract_event.rs

```diff
@@ -15,12 +15,36 @@ use proptest_derive::Arbitrary;
 use serde::{Deserialize, Serialize};
 use std::{convert::TryFrom, ops::Deref};
 
+pub trait ReadWriteEvent {
```

Comment:
Just to check my understanding, we basically can get bytes, patch them, and then write back using `update_event_data`?

Reply:
Yes. We can get the event data using `get_event_data`. We can patch it using `update_event_data`.

Reply:
This is a good, almost self-documenting trait interface, but I'd add some background comment to the trait, i.e. explain that sometimes we need to patch contents of emitted events, which is what the trait is for and allows.

Reply:
Done.